app/app/explore/components/AddLiquidityModal.tsx

```diff
@@ -105,21 +118,34 @@ export default function AddLiquidityModal({
     }
   }, [utxoChain, tokenBalance]);
 
-  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const value = e.target.value.replace(/[^0-9.]/g, "");
-    if (
-      value === "" ||
-      (!isNaN(parseFloat(value)) && parseFloat(value) <= assetBalance)
-    ) {
-      setAssetAmount(value);
-    }
+  const handleValueChange = (values: any) => {
```

Comment:
values should be type `NumberFormatValues` https://github.com/s-yadav/react-number-format/blob/master/src/types.ts#L11C18-L11C36